ballerina/types.bal

```diff
@@ -123,5 +133,47 @@ public enum Protocol {
     TLS
 }
 
+# The stream iterator object that is used to iterate through the stream messages.
+#
+class StreamIterator {
+    private boolean isClosed = false;
+
+    # Returns the next message in the stream.
+    # 
+    # + return - `record{|Message value;|}` or else `error?` if the stream is closed or any error occurred while retrieving the next message
+    public isolated function next() returns record{|Message value;|}|error? {
+        if self.isClosed {
+            return error Error("Stream is closed. Therefore, no operations are allowed further on the stream.");
+        }
+        Message|error? result = nextResult(self);
+        if result is Message {
+            return {value: result};
+        }
+        return result;
+    }
+
+    # Closes the stream.
+    # 
+    # + return - `error` if any error occurred while closing the stream or else `()`
+    public isolated function close() returns error? {
+        if !self.isClosed {
+            self.isClosed = true;
+            return closeStream(self);
+        } else {
+            return error Error("Stream is closed. Therefore, no operations are allowed further on the stream.");
+        }
+    }
+}
+
+isolated function nextResult(StreamIterator iterator) returns Message|error? =
+@java:Method {
+    'class: "io.ballerina.stdlib.mqtt.client.ClientActions"
+} external;
+
+isolated function closeStream(StreamIterator iterator) returns error? =
+@java:Method {
+    'class: "io.ballerina.stdlib.mqtt.client.ClientActions"
+} external;
```

Comment:
Lets move these inside the stream-iterator class bcos the functionality is for that particular class.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-mqtt/pull/9/commits/041d5b9016d02da09f39901fe4d4970544edfb25